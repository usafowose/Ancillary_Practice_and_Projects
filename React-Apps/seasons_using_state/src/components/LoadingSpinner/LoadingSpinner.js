import React from 'react'


const LoadingSpinner = (props) => {
  return (
    <div className='ui active dimmer' >
      <div className='ui big text loader' data-testid='LoadingMessage'>{props.message}</div>
    </div>
  )
}

LoadingSpinner.defaultProps = {
  message: `Loading...`
}

export default LoadingSpinner; 