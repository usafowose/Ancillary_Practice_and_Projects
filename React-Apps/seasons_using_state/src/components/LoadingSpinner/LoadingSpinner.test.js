import React from 'react';
import ReactDOM from 'react-dom';
import LoadingSpinner from './LoadingSpinner';

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'


afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<LoadingSpinner />, div)
});

it('renders LoadingSpinner correctly', () => {
  const elem = render(<LoadingSpinner message='Hello User' />);
  expect(elem.getByTestId('LoadingMessage')).toHaveTextContent('Hello User')
});

it('renders LoadingSpinner correctly', () => {
  const elem = render(<LoadingSpinner message='Goodbye User' />);
  expect(elem.getByTestId('LoadingMessage')).toHaveTextContent('Goodbye User')
});

// SnapShot Testing
import renderer from 'react-test-renderer';

it('matches snapshot 1', () => {
  const tree = renderer.create(<LoadingSpinner message='Testing Snapshot' />).toJSON();
  expect(tree).toMatchSnapshot();
})
it('matches snapshot 2', () => {
  const tree = renderer.create(<LoadingSpinner message='I am Not A Button, So Dont Push Me' />).toJSON();
  expect(tree).toMatchSnapshot();
})