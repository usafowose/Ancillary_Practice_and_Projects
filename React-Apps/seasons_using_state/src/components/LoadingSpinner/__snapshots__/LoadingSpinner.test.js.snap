// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="ui active dimmer"
>
  <div
    className="ui big text loader"
    data-testid="LoadingMessage"
  >
    Hi 
    Testing Snapshot
  </div>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="ui active dimmer"
>
  <div
    className="ui big text loader"
    data-testid="LoadingMessage"
  >
    Hi 
    I am Not A Button, So Dont Push Me
  </div>
</div>
`;
